regression/reg_task.py

```diff
@@ -122,11 +123,69 @@ def train_step(self, train_data):
 
         self.epoch_loss = self.loss_info.item()
 
-    def evaluate(self, x_test, n_samples=0):
+    def evaluate(self, x_test):
         self.net.eval()
         with torch.no_grad():
             y_test = self.net(x_test.to(DEVICE)).detach().cpu().numpy()
             return y_test
 
+    def log_progress(self, step):
+        write_loss(self.writer, self.loss_info, step)
+
+class MCDropout_Regression():
+    def __init__(self, label, parameters):
+        super().__init__()
+        self.writer = SummaryWriter(comment=f"_{label}_training")
+        self.label = label
+        self.lr = parameters['lr']
+        self.hidden_units = parameters['hidden_units']
+        self.mode = parameters['mode']
+        self.batch_size = parameters['batch_size']
+        self.num_batches = parameters['num_batches']
+        self.test_samples = parameters['test_samples']
+        self.x_shape = parameters['x_shape']
+        self.y_shape = parameters['y_shape']
+        self.save_model_path = f'{parameters["save_dir"]}/{label}_model.pt'
+        self.best_loss = np.inf
+        self.init_net(parameters)
+    
+    def init_net(self, parameters):
+        if not os.path.exists(parameters["save_dir"]):
+            os.makedirs(parameters["save_dir"])
+
+        model_params = {
+            'input_shape': self.x_shape,
+            'classes': self.y_shape,
+            'batch_size': self.batch_size,
+            'hidden_units': self.hidden_units,
+            'mode': self.mode
+        }
+        self.net = MLP_Dropout(model_params).to(DEVICE)
+        self.optimiser = torch.optim.Adam(self.net.parameters(), lr=self.lr)
+        self.scheduler = torch.optim.lr_scheduler.StepLR(self.optimiser, step_size=5000, gamma=0.5)
```

Comment:
sorry to be pedantic, step_size=500 more apt given regression epochs

Reply:
thx, fixed!